### DISCUSSION TITLE
search for autoimport extension for python

### DISCUSSION THREAD

**Discussion by hakanaku1234** [coder/code-server#3182] [Q&A]
hello, is there any python autoimport extension can work with latest code-server thx!

**Comment by jsjoeio** [comment]
Hmm...I haven't tried any myself. Is there one you want to use and it's not working?

You can also check the [Open VSX Registry](https://open-vsx.org/?search=python%20autoimport&category=&sortBy=relevance&sortOrder=desc)

**Comment by nooobcoder** [comment]
@hakanaku1234 you can use **Pylance** by Microsoft. It is a very smart Python linter, currently in Preview so you might face tiny issues. [Pylance](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance)

Go to the marketplace for visual studio and download the extension (vsix) file.
![image](https://user-images.githubusercontent.com/50350828/116023001-eee92b00-a668-11eb-83a9-68ca8bec002c.png)
After this, upload the extension to `code-server` and then install it by right clicking on it. `Select install from VSIX` from the right click context menu.
![image](https://user-images.githubusercontent.com/50350828/116023118-2d7ee580-a669-11eb-946b-b592ba423e5c.png)


**Also, if you feel I have answered your question and your query has been resolved, then please mark the discussion as marked, this would help the moderators to mark discussions that do not require to be discussed farther.**

Thank You,
from **Ankur Paul** 🇮🇳

**Reply by hakanaku1234** [reply]
thanks for your advice, this extension looks awesome, but in my situation, the `AutoImport` mechanism not works as expected :( 

I have changed language server to Pylance and reload window, `AutoComplete` works perfectly, but autoimport part didn't, 
should I do something more to make code-server support this part? thanks!!

**Reply by nooobcoder** [reply]
@hakanaku1234 thanks for reporting back, I checked and seems like Pylance does not work for code-server at all. I am creating an issue regarding the same for the developers to get notified about this.

**Reply by nooobcoder** [reply]
@hakanaku1234 i have mentioned this in an issue #3216 Lets hope for the best and a patch for this comes out soon. Thanks for letting me know this.
